### DISCUSSION TITLE
Thoughts on various fetch_* methods for DuckDBPyRelation

### DISCUSSION THREAD

**Discussion by wisp3rwind** [duckdb/duckdb#4091] [Ideas]
Hi,

currently, `DuckDBPyRelation` offers some convenience methods like `rel.fetchone()` which is essentially a shorthand for `rel.execute().fetchone()`. However, not all `fetch_*` methods of `DuckDBPyResult` are mirrored like this, and the naming of `Result` and `Relation` methods differs.

- Would you take a PR that adds the missing methods (for example, `fetchnumpy`) to `DuckDBPyRelation`?
- What is the history behind the method names? I find it quite confusing to have `DuckDBPyResult.fetchdf` and `DuckDBPyResult.fetch_df`, `DuckDBPyRelation.arrow`, `DuckDBPyRelation.to_arrow_table` and `DuckDBPyResult.fetch_arrow_table`, etc. Would it make sense to adjust the names here (e.g. rename all the methods to `fetch_*` and match `DuckDBPyResult` and `DuckDBPyRelation` methods, given that this is still duckdb 0.x.y and a stable API is not really guaranteed?

Thanks!

**Comment by Alex-Monahan** [comment]
Hmm, we have taken a "whichever you like" approach thus far. I am a fan of the simple `.df()` and `.arrow()` methods myself! Including all methods on both and aligning the names would be good though.
